cmd/private-org-peribolos-sync/main.go

```diff
@@ -0,0 +1,254 @@
+	if err := secretAgent.Start([]string{o.github.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+	gc, err := o.github.GitHubClient(secretAgent, false)
+	if err != nil {
+		logger.WithError(err).Fatal("Error getting GitHub client.")
+	}
+
+	orgRepos, err := getOrgReposWithOfficialImages(o.releaseRepoPath, o.includeReposSet)
+	if err != nil {
+		logger.WithError(err).Fatal("couldn't get the list of org/repos that promote official images")
+	}
+
+	peribolosRepos := generateRepositories(gc, orgRepos, logger)
+	peribolosConfigByOrg := peribolosConfig.Orgs[o.destOrg]
+	peribolosConfigByOrg.Repos = peribolosRepos
+	peribolosConfig.Orgs[o.destOrg] = peribolosConfigByOrg
+
+	out, err := yaml.Marshal(peribolosConfig)
+	if err != nil {
+		logrus.WithError(err).Fatalf("%s failed to marshal output.", o.peribolosConfig)
+	}
+
+	if err := ioutil.WriteFile(o.peribolosConfig, out, 0666); err != nil {
+		logrus.WithError(err).Fatal("Failed to write output.")
+	}
+}
+
+func generateRepositories(gc gitHubClient, orgRepos map[string]sets.String, logger *logrus.Entry) map[string]org.Repo {
+	peribolosRepos := make(map[string]org.Repo)
+
+	for orgName, repos := range orgRepos {
+		for repo := range repos {
+			logger.WithFields(logrus.Fields{"org": orgName, "repo": repo}).Info("Processing repository details...")
+
+			fullRepo, err := gc.GetRepo(orgName, repo)
+			if err != nil {
+				logger.WithError(err).Fatal("couldn't get repo details")
+			}
+
+			peribolosRepos[fullRepo.Name] = pruneRepoDefaults(org.Repo{
+				Description:      &fullRepo.Description,
+				HomePage:         &fullRepo.Homepage,
+				Private:          &fullRepo.Private,
+				HasIssues:        &fullRepo.HasIssues,
+				HasProjects:      &fullRepo.HasProjects,
+				HasWiki:          &fullRepo.HasWiki,
+				AllowMergeCommit: &fullRepo.AllowMergeCommit,
+				AllowSquashMerge: &fullRepo.AllowSquashMerge,
+				AllowRebaseMerge: &fullRepo.AllowRebaseMerge,
+				Archived:         &fullRepo.Archived,
+				DefaultBranch:    &fullRepo.DefaultBranch,
+			})
+		}
+	}
+
+	return peribolosRepos
+}
+
+func getOrgReposWithOfficialImages(releaseRepoPath string, includeRepos map[string]sets.String) (map[string]sets.String, error) {
+	ret := make(map[string]sets.String)
+
+	callback := func(c *api.ReleaseBuildConfiguration, i *config.Info) error {
+		if !promotion.BuildsOfficialImages(c) {
+			if repo, ok := includeRepos[i.Org]; !ok && !repo.Has(i.Repo) {
+				return nil
+			}
+		}
+
+		if _, ok := ret[i.Org]; !ok {
+			ret[i.Org] = sets.NewString(i.Repo)
+		} else {
+			ret[i.Org].Insert(i.Repo)
+		}
+
+		return nil
+	}
+
+	if err := config.OperateOnCIOperatorConfigDir(filepath.Join(releaseRepoPath, config.CiopConfigInRepoPath), callback); err != nil {
+		return ret, fmt.Errorf("error while operating in ci-operator configuration files: %v", err)
+	}
+
+	return ret, nil
+}
+
+// pruneRepoDefaults finds values in org.Repo config that matches the default
```

Comment:
This seems like something we need to share with upstream dump code, let's import it?

Reply:
I was hoping that too but it seems that this code exist in the main package of peribolos. I didn't want to move it and expose it just to use this once.

Reply:
Nope, we should *always* move and import